x-pack/plugins/actions/server/integration_tests/connector_types.test.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TestElasticsearchUtils, TestKibanaUtils } from '@kbn/core-test-helpers-kbn-server';
+import { ActionTypeRegistry } from '../action_type_registry';
+import { setupTestServers } from './lib';
+
+jest.mock('../action_type_registry', () => {
+  const actual = jest.requireActual('../action_type_registry');
+  return {
+    ...actual,
+    ActionTypeRegistry: jest.fn().mockImplementation((opts) => {
+      return new actual.ActionTypeRegistry(opts);
+    }),
+  };
+});
+
+const connectorTypes: string[] = [
+  '.email',
+  '.index',
+  '.pagerduty',
+  '.swimlane',
+  '.server-log',
+  '.slack',
+  '.slack_api',
+  '.webhook',
+  '.cases-webhook',
+  '.xmatters',
+  '.servicenow',
+  '.servicenow-sir',
+  '.servicenow-itom',
+  '.jira',
+  '.resilient',
+  '.teams',
+  '.torq',
+  '.opsgenie',
+  '.tines',
+  '.gen-ai',
+  '.bedrock',
+  '.d3security',
+  '.sentinelone',
+];
+
+describe('Connector type config checks', () => {
+  let esServer: TestElasticsearchUtils;
+  let kibanaServer: TestKibanaUtils;
+  let actionTypeRegistry: ActionTypeRegistry;
+
+  beforeAll(async () => {
+    const setupResult = await setupTestServers();
+    esServer = setupResult.esServer;
+    kibanaServer = setupResult.kibanaServer;
+
+    const mockedActionTypeRegistry = jest.requireMock('../action_type_registry');
+    expect(mockedActionTypeRegistry.ActionTypeRegistry).toHaveBeenCalledTimes(1);
+    actionTypeRegistry = mockedActionTypeRegistry.ActionTypeRegistry.mock.results[0].value;
+  });
+
+  afterAll(async () => {
+    if (kibanaServer) {
+      await kibanaServer.stop();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+  });
+
+  test('ensure connector types list up to date', () => {
+    expect(connectorTypes).toEqual(actionTypeRegistry.getAllTypes());
```

Comment:
Ahh okay, test will fail if new `connectorType` not added. 👍 
Still mock file could be useful if `connectorTypes` used in multiple test files. Up to you 🙂

Reply:
Added :)